tests/test_load_yaml_dict.py

```diff
@@ -250,6 +256,29 @@ def test_constraints_respected(self) -> None:
             "attrs >=20",
         ]
 
+    def test_constraints_respected_with_pip(self) -> None:
```

Comment:
Nice!

Did you see this test failing, before the fix in `devenv.py` was applied?

Reply:
Yep. It was failing.
This is the error without the fix:
```
self = <tests.test_load_yaml_dict.TestConstraints object at 0x0000024D8C643290>

    def test_constraints_respected_with_pip(self) -> None:
        """
        Constraints are declared as dependency if they are explicitly declared.
        """
        data = {
            "dependencies": ["attrs >19", "boltons", "pytest>=6", {"pip": ["lupa >= 1.14"]}],
            "constraints": ["pytest >7", "attrs >=20", "requests <2"],
        }
>       process_constraints_into_dependencies(data)

tests\test_load_yaml_dict.py:267:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
src\conda_devenv\devenv.py:351: in process_constraints_into_dependencies
    PackageSpecifier.parse(dep).name for dep in yaml_dict["dependencies"] #if isinstance(dep, str)
src\conda_devenv\devenv.py:444: in parse
    m = re.match(cls._PACKAGE_PATTERN, specifier)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

pattern = re.compile('^(?P<channel>[a-z0-9_\\-/.]+::)?(?P<package>[a-z0-9_\\-.]+)\\s*(?P<version>.*)$', re.IGNORECASE), string = {'pip': ['lupa >= 1.14']}, flags = 0

    def match(pattern, string, flags=0):
        """Try to apply the pattern at the start of the string, returning
        a Match object, or None if no match was found."""
>       return _compile(pattern, flags).match(string)
E       TypeError: expected string or bytes-like object, got 'dict'

..\..\envs\devenv\Lib\re\__init__.py:167: TypeError